Initial Domaintools Integration

wesleya opened the pull request:
<!-- Type of change
Please label this PR with one of the following labels, depending on the scope of your change:
- Bug
- Enhancement
- Breaking change
- Deprecation
-->

Initial integration for DomainTools Newly Observed Domains feed.

Please explain:

- WHAT: Initial datastream, ingest pipeline, fields for DomainTools Newly Observed Domains feed.
- WHY:  First DomainTools feed integration

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [x] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist

- [ ] Newly Observed Domains feed

## How to test this PR locally

Install and configure DomainTools integration. Search for feed results: `data_stream.dataset: "domaintools.nod_feed"`

wesleya commented:
Hi @andrewkroh I went ahead and signed the CLA, thank you!

andrewkroh commented on packages/domaintools/data_stream/nod_feed/fields/fields.yml:
```diff
@@ -0,0 +1,23 @@
+        - name: domain
+          type: keyword
+          description: >
+            The Domain.
```
Are these going to be [eTLDs](https://developer.mozilla.org/en-US/docs/Glossary/eTLD) only or any domain observed anywhere?

wesleya replied:
documentation has been updated. no eTLDS. Apex-level domains (e.g. example.com but not www.example.com) that we observe for the first time, and have not observed previously with our global DNS sensor network.

andrewkroh commented on packages/domaintools/docs/README.md:
```diff
@@ -0,0 +1,230 @@
+
+The `nod_feed` data stream provides events from [DomainTools Newly Observed Domains Feed](https://www.domaintools.com/products/threat-intelligence-feeds/).
+
+**Exported fields**
```
This field table and the sample event should be generated from a template that is placed in the `_build/` dir (see other integrations as an example).

wesleya replied:
Apologies, I wasn't able to figure this one out. I’m not sure what how to generate the sample event and field table here? I don’t see any template in `integrations/build` directory. Am I in the wrong place? 

I used the ti_recordedfuture README as an example. And just replaced the fields we export by hand. Please let me know what I'm missing here.

andrewkroh reviewed (commented):
I took quick (non-thorough) look and left a few comments. I'm asking to find the right internal team to help review this.

The integration kind of sounds like it should be treated as a threat intel integration with naming to match (e.g. ti_domaintools). Is that the general use case for this data?

This will need system and pipeline [testing](https://www.elastic.co/guide/en/integrations-developer/current/testing-and-validation.html).

kcreddy reviewed (commented):
All of our existing threat intel integrations have a prefix `ti_`. 
I would suggest you rename the integration to `ti_domaintools`.

For threat intel indicators, we also want to support an expiration.
You can take a look at other `ti_*` integrations how it is done. [Example](https://github.com/elastic/integrations/tree/main/packages/ti_threatconnect).

Essentially you need to setup a [latest transform](https://www.elastic.co/guide/en/elasticsearch/reference/current/transform-overview.html#latest-transform-overview) and expire the indicators based on a [time period](https://github.com/elastic/integrations/blob/main/packages/ti_recordedfuture/elasticsearch/transform/latest_ioc/transform.yml#L31-L33) or based on a [data point](https://github.com/elastic/integrations/blob/main/packages/ti_threatconnect/elasticsearch/transform/latest/transform.yml#L29-L30).

kcreddy commented:
/test

elasticmachine commented:
Pinging @elastic/security-service-integrations (Team:Security-Service Integrations)

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/19933)
* Commit: cbb8b5dcbab457b0b4d0d2536cfb36af734f94df

### Failed CI Steps
* [:white_check_mark: Check go sources](https://buildkite.com/elastic/integrations/builds/19933#01941b75-0ad5-4b1c-a963-1c5eb9c09b9b)

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/19933","number":19933,"commit":"cbb8b5dcbab457b0b4d0d2536cfb36af734f94df"}],"number":19933}
buildkite-pr-comment-->

wesleya commented on packages/domaintools/docs/README.md:
```diff
@@ -0,0 +1,230 @@
+
+### Newly Observed Domains (NOD) Feed 
+
+The `nod_feed` data stream provides events from [DomainTools Newly Observed Domains Feed](https://www.domaintools.com/products/threat-intelligence-feeds/).
```
@andrewkroh I've updated the README with a link to https://docs.domaintools.com/feeds/realtime/. This is a placeholder for where our public facing documentation for the API will live. I've attached a private preview to use for this code review:

[DomainTools-Realtime_Threat_Intelligence_Feeds[Technology_Preview].pdf](https://github.com/user-attachments/files/18497438/DomainTools-Realtime_Threat_Intelligence_Feeds.Technology_Preview.pdf)

Apologies for the the un finalized documentation, we are developing this integration in parallel with the API.

wesleya commented on packages/domaintools/data_stream/nod_feed/agent/stream/httpjson.yml.hbs:
@andrewkroh I've converted the httpjson input to a CEL input. Thank you!

wesleya commented on packages/domaintools/data_stream/nod_feed/fields/fields.yml:
```diff
@@ -0,0 +1,23 @@
+        - name: timestamp
+          type: keyword
+          description: >
+            The feed.
```
documentation has been updated. You are correct, first observed through passive DNS.

wesleya commented:
@kcreddy I've renamed the integration to ti_domaintools. I've also taken a first pass at adding a latest transform to support expiration. It seems to be working as expected. I've also addressed the feedback from @andrewkroh. 

Can you please take another look and let me know if we're missing anything. Appreciate your patience and support as first-timers working with Elastic, thank you!

wesleya commented:
@kcreddy @andrewkroh just checking in on this. Anything else needed on our side right now?

As always, appreciate the patience as elastic first-timers, thank you!

kcreddy commented:
/test

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/21258)
* Commit: 7a00771f33182f5a2cd41ecce77f998b708af991

### Failed CI Steps
* [:white_check_mark: Check go sources](https://buildkite.com/elastic/integrations/builds/21258#0194b5b2-e631-4553-a19d-b3af355953b4)

### History
* :broken_heart: [Build #19933](https://buildkite.com/elastic/integrations/builds/19933) failed cbb8b5dcbab457b0b4d0d2536cfb36af734f94df

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21258","number":21258,"commit":"7a00771f33182f5a2cd41ecce77f998b708af991"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/19933","number":19933,"commit":"cbb8b5dcbab457b0b4d0d2536cfb36af734f94df"}],"number":21258}
buildkite-pr-comment-->

kcreddy commented on packages/ti_domaintools/data_stream/nod_feed/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,55 @@
+  session_id: DomainToolsElasticSID
+  app_name: elastic_feeds
+  app_partner: elastic
+  app_version: 1.0.0
```
Does this need to be changed every time a new version is released?
Say we fix our first bug and upgrade from `0.1.0` to `0.1.1` and forgot to modify this value in CEL program. I believe the API still works?

wesleya replied:
Yes it will still work. The `app_version` is just for logging purposes on our side

kcreddy commented on packages/ti_domaintools/data_stream/nod_feed/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,55 @@
+  session_id: DomainToolsElasticSID
+  app_name: elastic_feeds
+  app_partner: elastic
+  app_version: 1.0.0
```
```suggestion
  app_version: 0.1.0
```

wesleya replied:
done!

kcreddy commented on packages/ti_domaintools/data_stream/nod_feed/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,55 @@
+                },
+            }
+        ).do_request().as(resp, (resp.StatusCode == 200) ?
+            {
+                "events": string(resp.Body).split("\n").filter(x,x!="").map(e,
+                    {
+                        "message": e,
+                    }
+                ),
+            }
```
Looks like CEL does not have support for API pagination. Does the API have pagination?
If so, its typically done using `want_more` in CEL.

Example: https://github.com/elastic/integrations/blob/main/packages/ti_opencti/data_stream/indicator/agent/stream/cel.yml.hbs#L75-L76

wesleya replied:
There is no pagination needed on this endpoint. We are planning to add more of our feed endpoints (which do have pagination) once this initial version is released. This is sort of our guinea pig since it's the easiest endpoint to use having no pagination.

kcreddy replied:
Thanks for clarification!

kcreddy commented on packages/ti_domaintools/manifest.yml:
```diff
@@ -0,0 +1,31 @@
+description: "The DomainTools NOD Feed provides real-time access to newly registered and observed domains, enabling proactive threat detection and defense."
+type: integration
+categories:
+  - custom
```
```suggestion
  - security
  - threat_intel
```

wesleya replied:
done!

kcreddy commented on packages/ti_domaintools/manifest.yml:
```diff
@@ -0,0 +1,31 @@
+format_version: 3.3.0
+name: ti_domaintools
+title: "DomainTools Real Time Unified Feeds"
+version: 1.0.0
```
```suggestion
version: 0.1.0
```

kcreddy commented on packages/ti_domaintools/data_stream/nod_feed/fields/fields.yml:
```diff
@@ -0,0 +1,24 @@
+      description: >
+        The feed.
+    - name: timestamp
+      type: keyword
```
```suggestion
      type: date
```

Use a `date` processor to convert to `date`. Example: https://github.com/elastic/integrations/blob/main/packages/mimecast/data_stream/siem_logs/elasticsearch/ingest_pipeline/v2_pipeline.yml#L4-L10

kcreddy replied:
Reference: https://www.elastic.co/guide/en/elasticsearch/reference/current/date-processor.html

wesleya replied:
Looking back into our docs, looks like the value we return here from our API is ISO-8601 UTC formatted timestamp. I believe that should work with the `date` type without need for a processor? Just the type switch from `keyword` to `date` should work here? lmk if I misunderstood here, thank you!

kcreddy replied:
> Just the type switch from keyword to date should work here?

Yes, that should also work. But I suggest you add explicit `date` processor because if there is any change to API date format, then instead of dropping document (due to mapping conflict), there will be an error logged in `error.message`

kcreddy commented on packages/ti_domaintools/data_stream/nod_feed/sample_event.json:
```diff
@@ -0,0 +1,58 @@
+    "domain": "bioalchemyglobal.com",
+    "timestamp": "2025-01-14T00:11:33Z"
+  },
+  "message": "{\"timestamp\":\"2025-01-14T00:11:33Z\",\"domain\":\"bioalchemyglobal.com\"}",
```
Not sure if this the full list of fields that API sends, but based on this, we could add few [`threat`](https://www.elastic.co/guide/en/ecs/current/ecs-threat.html) ECS fields.

1. [`threat.indicator.type`](https://www.elastic.co/guide/en/ecs/current/ecs-threat.html#field-threat-indicator-type)
2. [`threat.indicator.name`](https://www.elastic.co/guide/en/ecs/current/ecs-threat.html#field-threat-indicator-name)
3. `threat.feed.*` fields. Example: https://github.com/elastic/integrations/blob/main/packages/ti_opencti/data_stream/indicator/elasticsearch/ingest_pipeline/default.yml#L40-L51

kcreddy replied:
Also please add these to make this feed come up in Elastic Security solution:
https://github.com/elastic/integrations/blob/main/packages/ti_opencti/data_stream/indicator/elasticsearch/ingest_pipeline/default.yml#L17-L28

wesleya replied:
done!

kcreddy commented on packages/ti_domaintools/data_stream/nod_feed/fields/fields.yml:
Add a new fields file to set `labels.is_ioc_transform_source: true` for the source indices. 
Example: https://github.com/elastic/integrations/blob/main/packages/ti_opencti/data_stream/indicator/fields/is-ioc-transform-source-true.yml

wesleya replied:
done!

kcreddy commented on packages/ti_domaintools/elasticsearch/transform/latest_nod/fields/ecs.yml:
```diff
@@ -0,0 +1,54 @@
+  type: keyword
+- name: threat.indicator.url.full
+  type: keyword
+# Below fields to be moved into base-fields.yml after kibana.version changed to >= 8.14 
+# Related to fix: https://github.com/elastic/kibana/pull/177608
+- name: event.module
+  type: constant_keyword
+  description: Event module.
+  value: ti_domaintools
+- name: event.dataset
+  type: constant_keyword
+  description: Event dataset.
+  value: ti_domaintools.nod_feed
```
You can delete this block.

wesleya replied:
done!

kcreddy commented on packages/ti_domaintools/elasticsearch/transform/latest_nod/fields/fields.yml:
This file should match with what you have under `data_stream/nod_feed/fields/fields.yml`

wesleya replied:
done!

kcreddy commented on packages/ti_domaintools/elasticsearch/transform/latest_nod/fields/ecs.yml:
```diff
@@ -0,0 +1,54 @@
+  type: keyword
+- name: event.type
+  type: keyword
+- name: related.hash
+  type: keyword
+- name: related.ip
+  type: ip
+- name: related.user
+  type: keyword
```
You can only define those ECS fields that are populated by the integration. So some of these fields that are not required can be deleted.

wesleya replied:
done!

kcreddy commented on packages/ti_domaintools/elasticsearch/transform/latest_nod/transform.yml:
```diff
@@ -0,0 +1,36 @@
+  managed: true
+  # Bump this version to delete, reinstall, and restart the transform during package.
+  # Version bump is needed if there is any code change in transform.
+  fleet_transform_version: 0.4.0
```
```suggestion
  fleet_transform_version: 0.1.0
```

wesleya replied:
done!

kcreddy reviewed (commented):
Hey @wesleya, sorry for the delay.

The CI is currently failing with error: 
```
Error: there is no owner for "packages/ti_domaintools" in ".github/CODEOWNERS"
```
Could you please add an entry [here](https://github.com/elastic/integrations/blob/main/.github/CODEOWNERS#L406)

wesleya commented on packages/ti_domaintools/LICENSE.txt:
done!

wesleya commented on packages/ti_domaintools/docs/README.md:
ah thank you for the explanation, that worked, done!

wesleya commented on packages/ti_domaintools/changelog.yml:
```diff
@@ -0,0 +1,6 @@
+  changes:
+    - description: Initial draft of the package
+      type: enhancement
+      link: todo
```
done!

wesleya commented on packages/ti_domaintools/data_stream/nod_feed/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,55 @@
+config_version: "2"
+interval: {{interval}}
+resource.url: {{api_url}}
```
done!

wesleya commented on packages/ti_domaintools/elasticsearch/transform/latest_nod/transform.yml:
```diff
@@ -0,0 +1,36 @@
+  time:
+    field: event.ingested
+    # After internal discussion, 24 hours is a valid default retention period for Recorded Future.
+    max_age: 24h
```
Discussing with our product lead, we wanted to know if it's possible to have default interval of 10m for pulling in the feed, but expire the indicators after 7d? Reasoning is to keep the feed close to real time, but leave folks time to see them? Or let us know if we're misunderstanding `max_age` here. Thank you!

kcreddy replied:
Got it. So, with current configuration:
interval: `10m`
max_age: `7d`

This would mean when these domain indicators are deleted inside your platform (or expired since no longer valid), and when the API is called next time, this indicator will not be fetched from there on. But since `max_age` is set to `7d`, this indicator will still be present in your feed in Elastic and only deleted after `7d`, does this seem okay?

wesleya commented:
@kcreddy I believe we've addressed your feedback. Please let me know if there are more changes to make, or if my requests around `max_age`, and `date` above don't make sense. Thank you!

kcreddy commented:
/test

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/21441)
* Commit: 8dfaf8258b2ea6a889d1d9c385f8786cdfa18897

### Failed CI Steps
* [:white_check_mark: Check go sources](https://buildkite.com/elastic/integrations/builds/21441#0194ca56-3fe9-47dc-8fe1-36f87f0303c6)

### History
* :broken_heart: [Build #21258](https://buildkite.com/elastic/integrations/builds/21258) failed 7a00771f33182f5a2cd41ecce77f998b708af991
* :broken_heart: [Build #19933](https://buildkite.com/elastic/integrations/builds/19933) failed cbb8b5dcbab457b0b4d0d2536cfb36af734f94df

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21441","number":21441,"commit":"8dfaf8258b2ea6a889d1d9c385f8786cdfa18897"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21258","number":21258,"commit":"7a00771f33182f5a2cd41ecce77f998b708af991"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/19933","number":19933,"commit":"cbb8b5dcbab457b0b4d0d2536cfb36af734f94df"}],"number":21441}
buildkite-pr-comment-->

kcreddy commented:
@wesleya, thanks for resolving comments. I will take a look.

Could you please add an entry into [`.github/CODEOWNERS`](https://github.com/elastic/integrations/blob/main/.github/CODEOWNERS#L403)

CI error: https://github.com/elastic/integrations/pull/12140#pullrequestreview-2582949031

kcreddy commented on packages/ti_domaintools/elasticsearch/transform/latest_nod/transform.yml:
```diff
@@ -0,0 +1,36 @@
+retention_policy:
+  time:
+    field: event.ingested
+    # After internal discussion, 7d is a valid default retention period for Recorded Future.
```
```suggestion
    # After internal discussion, 7d is a valid default retention period for Domain Tools.
```

wesleya replied:
done!

kcreddy commented on packages/ti_domaintools/data_stream/nod_feed/manifest.yml:
```diff
@@ -0,0 +1,48 @@
+title: "DomainTools Real Time Unified Feeds"
+type: logs
```
You will need to add an ILM policy (only for the transform's source indices) and reference it here.
Because the transform sends all your latest indicators from source to destination indices, the destination indices are your current source of truth (latest indicators) and you can aggressively delete you source indices to prevent data being present in multiple places and optimise storage. This ILM policy helps with that.

Example:
1. Add ILM policy file: https://github.com/elastic/integrations/blob/main/packages/ti_recordedfuture/data_stream/threat/elasticsearch/ilm/default_policy.json
2. Reference in data-stream manifest.yml https://github.com/elastic/integrations/blob/main/packages/ti_recordedfuture/data_stream/threat/manifest.yml#L3
3. For Elastic Serverless deployments, ILM file in step 1 doesn't work. You will need another file `lifecycle.yml` placed inside `nod_feed` directory to apply that ILM policy.
Example: https://github.com/elastic/integrations/blob/main/packages/ti_recordedfuture/data_stream/threat/lifecycle.yml

wesleya replied:
done!

kcreddy commented on packages/ti_domaintools/manifest.yml:
```diff
@@ -0,0 +1,32 @@
+        title: "DomainTools Newly Observed Domains"
+        description: "DomainTools Newly Observed Domains provides real-time access to newly registered and observed domains, enabling proactive threat detection and defense"
+owner:
+  github: elastic/integrations
```
```suggestion
  github: elastic/security-service-integrations
```

https://github.com/elastic/integrations/blob/main/packages/ti_recordedfuture/manifest.yml#L41

wesleya replied:
done!

wesleya commented:
@kcreddy I've addressed your feedback comments. Thank you again for the help! 

I also noticed we now have merge conflicts with this message in our fork of the `elastic/integrations` repo today: 
> This branch is 5979 commits ahead, 6006 commits behind elastic/integrations:main.

With the only option in github's sync to "Discard 5979 commits". Should I go ahead and sync our fork with this option and merge into our `DomainTools:domaintools` branch? Looked a little odd so just wanted to check with your recommendation before I touched anything. Thank you!

![Screenshot 2025-02-04 at 8 25 44 AM](https://github.com/user-attachments/assets/1eb40817-f27d-4cdf-a4ec-9d8263caec11)

kcreddy commented:
@wesleya, unfortunately it has impacted all PRs currently open. 
Here are the instructions put forward by my team to resolve it (read in md).

```
To migrate a set of commits from one branch to another where the a rebase will fail due to conflicts and the number of conflicts are too great to reasonably resolve, there are a number of approaches:

Options:

1. in the case that only a small number of commits exist in the source branch, just cherry pick the commits to the new branch
2. if you have git 1.7.2+ you can cherry-pick a range of commits: `git cherry-pick OLDER^..NEWER`
3. if you have an earlier version or have not yet joined the century of the fruit bat, you can use `git format-patch` and then `git am` to simulate the ranged cherry-pick

Process:

- Identify the old (pre-fix) root of the PR branch. This is where the PR branches from the main branch in the pre-fix history. I checkout a new branch, `old_main`, for clarity, but you can just refer to this by sha.
- In all cases of the three options above, you should checkout a new branch at the current (fixed) main with a name related to the name of your PR branch; I just add a `_temp` suffix; `git checkout main && git checkout -b <original_branch>_temp`.
- Blat the commits from your original branch onto the new temp branch:
    1. (single commit): `git cherry-pick <original_branch>` or whichever GUI flavour you prefer.
    2. (multi-commit cherry-pick): `git cherry-pick old_main..<original_branch>`.
    3. (patches): `git checkout <original_branch>`, `git format-patch old_main`, `git checkout <original_branch>_temp`, and then `git am *.patch` (assuming no other .patch files are in the cwd).
- Rename the old branch as a safety measure: `git branch -m <original_branch> <original_branch>_orig`.
- Rename the new branch to enable a force push: `git branch -m <original_branch>_temp <original_branch>`.
- Force push the changes: `git push -f <upstream> <original_branch>`.
- When you have completed your fix-up, `git branch -D old_main` if you made one.

Caveats:
- I never use `git merge` to bring main back into a PR's chain of commits, so my history is always linear. You may see some conflicts if you do `git merge main` into your PRs, I don't know.
- If things fail, remember that you have made checkpoint branches that you can `git reset --hard` to.
```

wesleya commented:
@kcreddy thank you for the guidance, I believe I've updated the branch and things are looking as expected, testing looks good as well. 

You might see 2 or 3 additional force push attempts previous to my final attempt. You can ignore those, it just took me a few tries to work things out correctly. Let me know if I'm missing anything, thank you!

kcreddy commented:
/test

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/21811)
* Commit: 23609c14714f828860f1a8bbb3a38a949376bc7f

### Failed CI Steps
* [:white_check_mark: Check go sources](https://buildkite.com/elastic/integrations/builds/21811#0194d78c-eaa9-49dc-813a-31c0edc3c083)

### History
* :broken_heart: [Build #21441](https://buildkite.com/elastic/integrations/builds/21441) failed 8dfaf8258b2ea6a889d1d9c385f8786cdfa18897
* :broken_heart: [Build #21258](https://buildkite.com/elastic/integrations/builds/21258) failed 7a00771f33182f5a2cd41ecce77f998b708af991
* :broken_heart: [Build #19933](https://buildkite.com/elastic/integrations/builds/19933) failed cbb8b5dcbab457b0b4d0d2536cfb36af734f94df

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21811","number":21811,"commit":"23609c14714f828860f1a8bbb3a38a949376bc7f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21441","number":21441,"commit":"8dfaf8258b2ea6a889d1d9c385f8786cdfa18897"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/21258","number":21258,"commit":"7a00771f33182f5a2cd41ecce77f998b708af991"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/19933","number":19933,"commit":"cbb8b5dcbab457b0b4d0d2536cfb36af734f94df"}],"number":21811}
buildkite-pr-comment-->

kcreddy commented:
There's a CI error unrelated to this package. https://github.com/elastic/integrations/pull/12630 should fix the CI error.
Once that is merged, CI needs to be run again.

wesleya commented:
@kcreddy I saw another force-push come in causing more conflicts on this PR. Should I repeat the resolution steps again? Or should I wait on #12630 or other changes to come in first?

kcreddy commented:
> Should I repeat the resolution steps again? Or should I wait on https://github.com/elastic/integrations/pull/12630 or other changes to come in first?

I will let you know as I get some updates. Right now the merging is blocked. Apologies for the inconvenience.

wesleya commented:
I just received this email from security@elastic.co: 

> We are working with GitHub support to clean up https://github.com/DomainTools/integrations/tree/domaintools. The GitHub engineering team has identified that you have an active branch against the fork of elastic/integrations. This prevents GitHub from taking action on the files and their references.
> 
> We kindly request that you delete the branch, delete and re-fork the repo, and please confirm when complete.

I will re-create this PR on a brand new branch on the re-forked repo. Thank you
